mesonbuild/mcompile.py

```diff
@@ -83,14 +120,22 @@ def run(options: 'argparse.Namespace') -> int:
 
         cmd = [runner, '-C', bdir.as_posix()]
 
+        if options.targets and options.clean:
+            raise MesonException('`target` and `clean` can\'t be used simultaneously')
```

Comment:
argparse has builtin functionality for this: https://docs.python.org/3/library/argparse.html#mutual-exclusion

Reply:
Thanks for the tip! Fixed

Reply:
This results in `error: argument TARGET: not allowed with argument --clean`, when invoking like this: `meson compile -C dir --clean`.
It seems that `TARGET` is always defined as present even it's just an empty default value.

Reply:
hmmmm, I'm not sure off the top of my head. It may not be possible to make a switch argument and positional argument mutally exclusive. You may need to either drop my suggestion or use something like `-t/--target=foo`.

I'm fine with whichever way you want to go. Though if decide to not use the mutually exclusive arguments, can we move the argument exclusive checking into common code so each backend function doesn't need to implement it?

Reply:
Removed `mutual_exclusion` and extracted the check to the common section.